parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -76,6 +67,31 @@ protected ParquetValueReader<T> createReader(
   protected abstract ParquetValueReader<T> createStructReader(
       List<Type> types, List<ParquetValueReader<?>> fieldReaders, Types.StructType structType);
 
+  protected ParquetValueReaders.PrimitiveReader<?> fixedReader(ColumnDescriptor desc) {
+    return null;
```

Comment:
I think this has to return a generic fixed reader here to match the existing behavior. Because this class is public, people may have created subclasses for other object models, and `iceberg-parquet` is a module that is covered by `revapi` so we make guarantees about API and behavior stability.

Reply:
Since this class is not widely used, could you also mark it deprecated so that we can make it package-private in 1.9.0?

Reply:
I didn't do it because, I can't move the date/time classes to `GenericReader`. It has to be in this class. But yeah, I got your point about behavior change.

Reply:
@ajantha-bhat, why can't the reader classes be moved if this returns an instance? The classes are currently `private` so there is no direct dependency on them. As long as an instance is returned, wouldn't it work?

Reply:
I see. I moved them back now. I didn't create extra factory methods for these existing classes after the movement as I felt it is an overkill and behavior shouldn't change for these generic readers and writers.